diff --git a/Android.mk b/Android.mk
old mode 100644
new mode 100755
index 844897b..3361641
--- a/Android.mk
+++ b/Android.mk
@@ -683,17 +683,17 @@ include $(BUILD_EXECUTABLE)
 #include $(BUILD_EXECUTABLE)
 #
 ########################
-#
-#local_target_dir := $(TARGET_OUT)/etc/wifi
-#
-#include $(CLEAR_VARS)
-#LOCAL_MODULE := wpa_supplicant.conf
-#LOCAL_MODULE_TAGS := user
-#LOCAL_MODULE_CLASS := ETC
-#LOCAL_MODULE_PATH := $(local_target_dir)
-#LOCAL_SRC_FILES := $(LOCAL_MODULE)
-#include $(BUILD_PREBUILT)
-#
+
+local_target_dir := $(TARGET_OUT)/etc/wifi
+
+include $(CLEAR_VARS)
+LOCAL_MODULE := wpa_supplicant.conf
+LOCAL_MODULE_TAGS := user
+LOCAL_MODULE_CLASS := ETC
+LOCAL_MODULE_PATH := $(local_target_dir)
+LOCAL_SRC_FILES := $(LOCAL_MODULE)
+include $(BUILD_PREBUILT)
+
 ########################
 
 endif # ifeq ($(WPA_BUILD_SUPPLICANT),true)
diff --git a/driver_wext.c b/driver_wext.c
old mode 100644
new mode 100755
index a2e4578..97630f1
--- a/driver_wext.c
+++ b/driver_wext.c
@@ -1366,7 +1366,8 @@ int wpa_driver_wext_get_scan_results(void *priv,
 			if (ap_num < max_size) {
 				results[ap_num].qual = iwe->u.qual.qual;
 				results[ap_num].noise = iwe->u.qual.noise;
-				results[ap_num].level = iwe->u.qual.level;
+				results[ap_num].level = iwe->u.qual.level - 256;
+				
 			}
 			break;
 		case SIOCGIWENCODE:
diff --git a/wpa_supplicant.conf b/wpa_supplicant.conf
old mode 100644
new mode 100755
index e34d49f..5d9a98d
--- a/wpa_supplicant.conf
+++ b/wpa_supplicant.conf
@@ -72,7 +72,7 @@ update_config=1
 # DACL (which will reject all connections). See README-Windows.txt for more
 # information about SDDL string format.
 #
-ctrl_interface=wlan0
+ctrl_interface=DIR=/data/system/wpa_supplicant GROUP=system
 
 # IEEE 802.1X/EAPOL version
 # wpa_supplicant is implemented based on IEEE Std 802.1X-2004 which defines
